### ISSUE TITLE
Computed size field in the Sprite structure

### ISSUE THREAD

**Issue by alexfacciorusso** [bevyengine/bevy#5724]
## What problem does this solve or what need does it fill?

Getting sprite sizes is nor ergonomic nor productive at the moment. 

## What solution would you like?

Adding a `computed_size: Option<Vec2>` or similar to the `Sprite` struct. That field will be updated from the sprite engine whenever it knows the size and image of the Sprite.

## What alternative(s) have you considered?

The alternative is every time we need the size of a sprite (or an image associated to one), to request the asset server, the image handle, checking that the image is loaded and then accessing to its size and potentially creating a resource or a component in the entity with the size for a system or systems to access.

**Comment by djeedai** [comment]
For context the Sprite pipeline currently get this info in the queue phase from the `GpuImage` here:

https://github.com/bevyengine/bevy/blob/aed3232e38e2d700cbb06b782ed7242d0684c244/crates/bevy_sprite/src/render/mod.rs#L456

**Comment by alexfacciorusso** [comment]
Also, in case of a custom size set on the Sprite, the computed size field would/should obviously be equal to that.

**Comment by htrojan** [comment]
I think some method for retrieving the image size would be really helpful.
When trying to implement collision detection for sprites with textures, the missing size information gets in the way. 
The official examples deal with collision detection by storing the size information in transform.scale and using this as a bounding box. But when loading an image into the sprite, the transform has to be (1, 1, 1). Otherwise the image itself will be scaled. My first intuition was that transform.scale would then be set by the sprite loader to the image dimensions, which is not the case. Therefore, some mechanism for retrieving the size information of loaded images to build at least some kind of external bounding box system on top would be great

**Comment by rparrett** [comment]
I just updated a discussion I posted shortly after Bevy 0.6's release on this topic: https://github.com/bevyengine/bevy/discussions/3620.

I agree that this is a bit of a usability issue for folks who are learning Bevy from our examples or just prototyping.

**Comment by sullyj3** [comment]
`TextureAtlas` could also use this.

**Comment by targrub** [comment]
The trick is when to set the value of that field.  When do we know the size of the image?  Are we at that point in a context in which we have the Sprite(s) that reference it?  If not, is there somewhere else in the code when it makes sense to store the image size on a Sprite?

**Comment by sullyj3** [comment]
I agree this wouldn't necessarily have to be a field, even a convenience method would make things a lot nicer. It is definitely tricky to figure out what the method should be on, given that you need access to the `Sprite`, `Transform`, and `Handle<Image>` to compute this. See this [relevant discussion](https://github.com/bevyengine/bevy/discussions/3620) for a code example of the current way to do it.

Imo computed fields introduce unnecessary opportunities for invalid states/desynchronization, so they're best avoided if at all possible. It's something you have to remember to keep updated in all the appropriate places when changing the code.
